chemprop/cli/train.py

```diff
@@ -46,26 +45,37 @@
 class TrainSubcommand(Subcommand):
     COMMAND = "train"
     HELP = "train a chemprop model"
+    parser = None
```

Comment:
I really like this idea and would like to expand it in a future PR. Currently the object model is that each subcommand "knows what additional args it will need" but that's a far cry from the simpler "knows the parser that it will need." I wonder if the latter is possible, but that's out of scope for this PR.